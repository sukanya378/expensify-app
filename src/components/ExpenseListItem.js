import React from "react";
import { NavLink } from "react-router-dom";

const ExpenseListItem = (props) => {
    const { id, description, amount, createdAt } = props.expense
    
    return (
        <div className="expense-list-item">
            <NavLink to={`/edit/${id}`} activeClassName="is-active"><h3>{description}</h3></NavLink>
            <div>{amount}</div>
            <div>{createdAt}</div>
            
        </div>
    )
}

export default ExpenseListItem;