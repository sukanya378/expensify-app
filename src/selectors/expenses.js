import moment from "moment";

const getVisibleExpenses = (expenses, { text, sortBy, startDate, endDate }) => {
    expenses =  expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt)
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase())

        return startDateMatch && endDateMatch && textMatch;
    })

    expenses.sort((a,b) => {
        if(sortBy === "amount") return b.amount - a.amount
        else return b.createdAt - a.createdAt
    })
    return expenses
}

export default getVisibleExpenses