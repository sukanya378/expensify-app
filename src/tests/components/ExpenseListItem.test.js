import React from "react";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses"
import ExpenseListItem from "../../components/ExpenseListItem";

test('should render Expense list item ', () => {
    const wrapper = shallow(<ExpenseListItem expense={expenses[0]} />)
    expect(wrapper).toMatchSnapshot()
})