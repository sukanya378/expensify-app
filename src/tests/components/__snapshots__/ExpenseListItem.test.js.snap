// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense list item  1`] = `
<div
  className="expense-list-item"
>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/edit/1"
  >
    <h3>
      rent
    </h3>
  </NavLink>
  <div>
    9000
  </div>
  <div>
    1686126588493
  </div>
</div>
`;
